### PR TITLE
CSS/SASS refactor

### PR DESCRIPTION
**Pull request by daviferreira**
This PR includes a couple of breaking changes I think, but I want to have a `medium-editor` namespace on all our css classes/attrs.

* `medium-element` becomes `medium-editor-element`
* `sticky-toolbar` becomes `medium-editor-sticky-toolbar`
* `stalker-toolbar` becomes `medium-editor-stalker-toolbar`

Maybe we should still keep the old classes for now? Not sure.

The rest was mainly creating components and organizing editor/toolbar components. There is more room for improvement, but this is a start.

### CODE REVIEW

**Comment by j0k3r** [comment]
If it's not too hard maybe we can keep old classes name and remove them when realizing v5

**Comment by phiggins42** [comment]
+1 @j0k3r .. these changes are pretty breaking if you had any use of anything like "the top level element is called `[data-medium-element]` (like a custom traverseup? or basically anything). also any custom themeing goes out the window. I vote keep this out until 5.0, or rather, use this as the first patch set towards 5.0, releaseing whatever is in right now as the last in the 4.x tree (or, even better, use proper branch strategy to allow the two to live simultaneously for a while)

**Comment by nmielnik** [comment]
Yeah, with 5.0 coming soon (hopefully very soon), I think we should either keep the old classes along with the new ones, or hold off on these changes until they can get lumped into the v5.0.0 release.

File: src/sass/components/_anchor-preview.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    top: 0;
+    word-break: break-all;
+    word-wrap: break-word;
+    visibility: hidden;
+    z-index: $z-toolbar;
+
+    a {
+        color: #fff;
+        display: inline-block;
+        margin: 5px 5px 5px 10px;
```
Comment (nchase) [inline src/sass/components/_anchor-preview.scss]: is there any reason these weren't condensed to `margin: 5px 5px 10px`?

File: src/sass/components/_anchor-preview.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+.medium-editor-anchor-preview {
+    font-family: $font-sans-serif;
+    font-size: 16px;
```
Comment (nchase) [inline src/sass/components/_anchor-preview.scss]: in web browsers, the `font-size` is equal to this value. Can this be a `font-size: inherit` instead in the interest of making this an easy drop-in solution on the CSS side?

File: src/sass/components/_placeholder.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.medium-editor-placeholder {
+    position: relative;
+
+    &:after {
+        content: attr(data-placeholder) !important;
```
Comment (nchase) [inline src/sass/components/_placeholder.scss]: what is this `!important` overriding?

**Comment by nchase** [comment]
this is going to be awesome for 5.x.x :thumbsup:

File: src/sass/components/_placeholder.scss (hunk 3 above)
Comment (daviferreira) [inline src/sass/components/_placeholder.scss]: I have no idea, but there must be a reason :P

File: src/sass/components/_anchor-preview.scss (hunk 2 above)
Comment (daviferreira) [inline src/sass/components/_anchor-preview.scss]: Same as toolbar, the main container. I think we should leave it for now, but font-sizes and margin/paddings is something I would like to improve.

**Pull request closed without merging**